### PR TITLE
rustup: Default to the beta channel

### LINKED ISSUE #21149: Update rustup.sh for beta
Currently it doesn't support channels. When the beta is released it should default to beta and support a `--channel` flag to switch back to nightly.

### PR DESCRIPTION
**Pull request by alexcrichton**
Switches rustup to using the beta channel by default. Includes #23824 for the implementation.

cc #20453
Closes #21149

### CODE REVIEW

**Comment by steveklabnik** [comment]
it's happening

File: src/etc/rustup.sh [hunk 1]
```diff
@@ -449,18 +450,28 @@ then
...
+            RUST_URL="${RUST_URL}/${CFG_DATE}"
+        fi
+
+        RUST_PACKAGE_NAME=rust-nightly
+        ;;
+    beta)
+        RUST_PACKAGE_NAME=rust-1.0.0-beta
+        ;;
+    *)
+        err "Currently nightly is the only supported release channel"
```
Comment (SimonSapin) [inline src/etc/rustup.sh]: Should this say "Currently nightly and beta are the only supported release channels"?

Comment (richo) [inline src/etc/rustup.sh]: Throwback from my original commit when I didn't know where the other artifacts came from

**Comment by richo** [comment]
Thanks for running with this!

**Comment by richo** [comment]
Could you cherry pick 8409d854565491615049d497bcc41748edd97ac0 from my branch as well? Seems a waste to actually PR (Can also PR)

**Comment by aturon** [comment]
@alexcrichton looks good to me! r=me once you decide whether to cherry pick.

**Pull request merged**